client/my-sites/guided-transfer/host-credentials-page/index.jsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Bluehost from './bluehost';
+import SiteGround from './siteground';
+import SectionHeader from 'components/section-header';
+
+class HostCredentialsPage extends Component {
+	static propTypes = {
+		hostSlug: PropTypes.string.isRequired,
+	};
+
+	state = { fieldValues: {} };
```

Comment:
I like this simplicity of using refs, but I think `state` is more appropriate here. A few reasons:

1. There's less to change if we do need to move the state into the Redux store later. React state is a simple JS object and can be totally replaced with a Redux `connect`, while `ref`s are very React specific. `setFieldValue` and `onFieldChange` could be connected to Redux state and actions, and the child components wouldn't need to know the difference.

2. Using refs in a single component is a nice way to grab values from uncontrolled components without boilerplate, but passing refs between multiple components doesn't seem like a good idea. Since the refs are references to actual component instances, these could change underneath us when React feels like it.

3. `host-credentials-page/index.jsx` doesn't need to know about the fields that are contained within the host specific pages. Its only purpose is to collect whatever data is required for a host, then submit it - it doesn't need to know the shape of that data.

> No more rendering with every keystroke.

React's virtual DOM/diff algorithms are great at avoiding this - they'll only render to the DOM if the markup is different to what's expected.

> Its not using state anymore.

I don't think there's anything inherently wrong with using React's `state`. It's unavoidable that there will be state somewhere, it's just a matter of whether or not we explicitly [control it](https://facebook.github.io/react/docs/forms.html#controlled-components). React's state might just have a bad rep because it was commonly used to store far too much complex app state, which we now manage with Flux/Redux.